tests/mtest/src/main/scala/tests/Compat.scala

```diff
@@ -0,0 +1,39 @@
+package tests
+
+import scala.meta.internal.semver.SemVer
+
+object Compat {
+
+  /**
+   * Cases might be described as:
+   *  - Starts swith: "3.0.1" -> value | "3.0" -> value | "3"
+   *  - Greater or equal: ">=3.0.0" -> value
+   */
+  def forScalaVersion[A](
+      scalaVersion: String,
+      cases: Map[String, A]
+  ): Option[A] = {
+    val (startsWith, gt) =
+      cases.partition({ case (v, _) => !v.startsWith(">=") })
+
+    val fromStartWith = startsWith.collectFirst({
+      case (v, a) if scalaVersion.startsWith(v) => a
+    })
+    matchesGte(scalaVersion, gt) orElse fromStartWith
+  }
+
+  private def matchesGte[A](
+      version: String,
+      gtCases: Map[String, A]
+  ): Option[A] = {
+    val parsed = SemVer.Version.fromString(version)
+    val less = gtCases
+      .map { case (v, a) =>
+        (SemVer.Version.fromString(v.stripPrefix(">=")), a)
+      }
+      .filter({ case (v, _) => parsed >= v })
```

Comment:
```suggestion
      .filter { case (v, _) => parsed >= v }
```

Reply:
thx for suggestions! applied all of them